### ISSUE TITLE
get.dataset does not work properly since 7.17.0 (zowe sdk)

### ISSUE THREAD

**Issue by kraabrsg** [zowe/zowe-cli#1813]
**Describe the bug**

When opening big files, sometimes ( and not reproduceable on our side ) this await of the dataset does not finish.
Just "awaiting" without returning a result
 ```
    public static async getDatasetString(dataset: string, session?: Session) {
        return (await Get.dataSet(session ?? this.session, this.encodeDatasetName(dataset), { encoding: Configuration.hostEncoding })).toString();
    }
```
Session is a static session:
![image](https://github.com/zowe/zowe-cli/assets/86772851/aec70e44-735a-4bc2-8dc1-58742aa21d88)

**Expected and actual results**

Return the string of the file...

**Describe your environment**

WIN10
Vscode 1.82.2
Depencies of the vscode extension, where get.dataset is called (any number from 7.17.x onwards seems to have the issue) :
```
	"dependencies": {
		"@types/xml2js": "^0.4.12",
		"@vscode/codicons": "^0.0.33",
		"@vscode/debugadapter": "^1.63.0",
		"@vscode/debugprotocol": "^1.63.0",
		"@zowe/imperative": "^5.18.1",
		"@zowe/zos-files-for-zowe-sdk": "7.16.4",
		"@zowe/zos-jobs-for-zowe-sdk": "^7.18.2",
		"@zowe/zos-tso-for-zowe-sdk": "^7.18.2",
		"@zowe/zosmf-for-zowe-sdk": "^7.18.2",
       ....
      }
```

Thanks!

**Comment by JTonda** [comment]
Related to #1805

**Comment by KevinLoesch1** [comment]
This issue is corrected with version [7.18.9](https://github.com/zowe/zowe-cli/pull/1838).
